packages/atomic/src/components/atomic-modal/atomic-modal.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  Component,
+  h,
+  State,
+  Prop,
+  Element,
+  Watch,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+  DeferUntilRender,
+} from '../../utils/initialization-utils';
+import {getFirstFocusableDescendant} from '../../utils/accessibility-utils';
+import {updateBreakpoints} from '../../utils/replace-breakpoint';
+import {once} from '../../utils/utils';
+
+/**
+ * When the modal is opened, the class `atomic-modal-open` is added to the body, allowing further customization.
+ *
+ * @part backdrop - The transparent backdrop hiding the content behind the modal.
+ * @part container - The modal's outermost container with the outline and background.
+ * @part header-wrapper - The wrapper around the header.
+ * @part header - The header at the top of the modal.
+ * @part header-ruler - The horizontal ruler underneath the header.
+ * @part body-wrapper - The wrapper around the body.
+ * @part body - The body of the modal, between the header and the footer.
+ * @part footer-wrapper - The wrapper with a shadow around the footer.
+ * @part footer - The footer at the bottom of the modal.
+ * @internal
+ */
+@Component({
+  tag: 'atomic-modal',
+  styleUrl: 'atomic-modal.pcss',
+  shadow: true,
+})
+export class AtomicModal implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() public host!: HTMLElement;
+
+  @State() public error!: Error;
+
+  @Prop({mutable: true}) source?: HTMLElement;
+  @Prop({reflect: true, mutable: true}) isOpen = false;
+  @Prop({mutable: true}) close: () => void = () => (this.isOpen = false);
+
+  @Event() animationEnded!: EventEmitter<never>;
+
+  private wasEverOpened = false;
+
+  @Watch('isOpen')
+  watchToggleOpen(isOpen: boolean) {
+    const modalOpenedClass = 'atomic-modal-opened';
+
+    if (isOpen) {
+      this.wasEverOpened = true;
+      document.body.classList.add(modalOpenedClass);
+      this.focusOnFirstElement();
+    } else {
+      document.body.classList.remove(modalOpenedClass);
+      this.focusOnSource();
+    }
+  }
+
+  @DeferUntilRender()
+  private focusOnFirstElement() {
+    getFirstFocusableDescendant(this.host)?.focus();
+  }
+
+  @DeferUntilRender()
+  private focusOnSource() {
+    this.source?.focus();
+  }
+
+  private updateBreakpoints = once(() => updateBreakpoints(this.host));
+
+  public componentWillLoad() {
+    this.wasEverOpened ||= this.isOpen;
+  }
+
+  public render() {
+    if (!this.wasEverOpened) {
+      return;
+    }
+
+    this.updateBreakpoints();
+
+    return (
+      <div
+        part="backdrop"
+        class={`fixed left-0 top-0 right-0 bottom-0 z-10 ${
+          this.isOpen ? 'opened' : ''
+        }`}
+        onClick={(e) => e.target === e.currentTarget && this.close()}
+      >
+        <atomic-focus-trap active={this.isOpen}>
+          <article
```

Comment:
As explained in the scrum, I think it would be a good time investment to do a little technical exploration into how other popular (and accessibility compliant) library do their HTML markup for a modal, ie: which semantic element they are using.

To see if there's some kind of repeating pattern for all of them, or if they all do it differently.

Reply:
Here are the samples I found from some libraries:

# Modal libraries with styling
## ionic (JS)
```html
<ion-app>
  <ion-modal is-open="true">
    <ion-content>Modal Content</ion-content>
  </ion-modal>
</ion-app>
```

## react-bootstrap (non-static)
```jsx
<Button variant="primary" onClick={handleShow}>
  Launch demo modal
</Button>
<Modal show={show} onHide={handleClose}>
  <Modal.Header closeButton>
    <Modal.Title>Modal heading</Modal.Title>
  </Modal.Header>
  <Modal.Body>Woohoo, you're reading this text in a modal!</Modal.Body>
  <Modal.Footer>
    <Button variant="secondary" onClick={handleClose}>
      Close
    </Button>
    <Button variant="primary" onClick={handleClose}>
      Save Changes
    </Button>
  </Modal.Footer>
</Modal>
```

# Modal libraries without styling
## react-modal
```jsx
<div>
  <button onClick={openModal}>Open Modal</button>
  <Modal
    isOpen={modalIsOpen}
    onAfterOpen={afterOpenModal}
    onRequestClose={closeModal}
    style={customStyles}
    contentLabel="Example Modal"
  >
    <h2 ref={(_subtitle) => (subtitle = _subtitle)}>Hello</h2>
    <button onClick={closeModal}>close</button>
    <div>I am a modal</div>
    <form>
      <input />
      <button>tab navigation</button>
      <button>stays</button>
      <button>inside</button>
      <button>the modal</button>
    </form>
  </Modal>
</div>
```

## styled-react-modal
```jsx
<div>
  <button onClick={toggleModal}>Click me</button>
  <StyledModal
    isOpen={isOpen}
    onBackgroundClick={toggleModal}
    onEscapeKeydown={toggleModal}>
    <span>I am a modal!</span>
    <button onClick={toggleModal}>Close me</button>
  </StyledModal>
</div>
```

## micromodal
```html
<script>
  MicroModal.init();
  MicroModal.show('modal-1');
</script>
<style>
  .modal {
    display: none;
  }

  .modal.is-open {
    display: block;
  }
</style>
<div id="modal-1" aria-hidden="true">
  <div tabindex="-1" data-micromodal-close>
    <div role="dialog" aria-modal="true" aria-labelledby="modal-1-title" >
      <header>
        <h2 id="modal-1-title">
          Modal Title
        </h2>
        <button aria-label="Close modal" data-micromodal-close></button>
      </header>
      <div id="modal-1-content">
        Modal Content
      </div>
    </div>
  </div>
</div>
```